pkg/controllers/dnsrecord/dnsrecord_controller.go

```diff
@@ -158,7 +158,7 @@ func (r *DNSRecordReconciler) deleteRecord(ctx context.Context, dnsRecord *v1alp
 
 	err = dnsProvider.Delete(dnsRecord, managedZone)
 	if err != nil {
-		if strings.Contains(err.Error(), "was not found") {
+		if strings.Contains(err.Error(), "was not found") || strings.Contains(err.Error(), "notFound") {
```

Comment:
Created follow on task for this https://github.com/Kuadrant/multicluster-gateway-controller/issues/458